### ISSUE TITLE
Escaped characters in keys and values

### ISSUE THREAD

**Issue by cmaumet** [gllmflndn/JSONio#1]
Hi @gllmflndn! The following example (from our discussion at: https://github.com/incf-nidash/nidmresults-spm/issues/40#issuecomment-287042869):
```
>> json = containers.Map('nidm_DesignMatrix/prov:atLocation','./DesignMatrix.csv');
>> jsonwrite(json)
```
returns:
```
ans =
 {"nidm_DesignMatrix\/prov:atLocation":".\/DesignMatrix.csv"}
```

Is there any particular reason for the slashes to be escaped? I was hoping we could get:
```
{"nidm_DesignMatrix/prov:atLocation": "./DesignMatrix.csv"}
```

I am happy to investigate this and send a PR but I would like to make sure I understand when the escapes are needed. 

Thank you in advance!

**Comment by gllmflndn** [comment]
From http://json.org/, it is my understanding that '/' should have a backslash escape. Both pass validation through [JSONLint](http://jsonlinthttp://stackoverflow.com/questions/1580647/json-why-are-forward-slashes-escaped.com/) though. It seems to be a frequently discussed topic, e.g.:
* http://stackoverflow.com/questions/1580647/json-why-are-forward-slashes-escaped
* https://github.com/esnme/ultrajson/issues/110

It's all taking place there:
* https://github.com/gllmflndn/JSONio/blob/master/jsonwrite.m#L172

**Comment by cmaumet** [comment]
Thank you @gllmflndn: I was not aware of this issue! From the thread you linked to, my understanding is that both notations are valid, and that:
```
 {"nidm_DesignMatrix\/prov:atLocation":".\/DesignMatrix.csv"}
```
is equivalent to
```
 {"nidm_DesignMatrix/prov:atLocation":"./DesignMatrix.csv"}
```

In [this document](http://www.ietf.org/rfc/rfc4627.txt) "control characters" are explicitly defined as  `U+0000`  through `U+001F` which does not include `/` (`U+002F`). 

Do you think it would make sense to use the non-escaped version by default? Or create an option to espace only if mandatory?

**Comment by gllmflndn** [comment]
JSON specification(s) are sometimes hard to follow, see:
http://seriot.ch/parsing_json.php#1
I just checked and MATLAB's `jsonencode` does not escape forward slashes so, for compatibility, I will just do the same here. That will solve the issue for you: how come it is a problem for you though, given that a reader would un-escape the escaped characters?
```
>> jsonread(jsonwrite(struct('a','./hello')))
ans = 
  struct with fields:
    a: './hello'
```

**Comment by gllmflndn** [comment]
Forward slashes are not escaped any more, see 3a81c3d, and this behaviour is now compatible with ``jsonencode``.
It would be possible to add an option as suggested in pull request #2 but implementation should instead use ``optregistry`` to minimise changes throughout the file - low priority though as I hope current version fixes the issue for you.
